src/Kernel/Magic/ViewMagic.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Jupyter.Core;
+using Microsoft.Jupyter.Core.Protocol;
+using Microsoft.Quantum.IQSharp.Core.ExecutionPathTracer;
+using Microsoft.Quantum.IQSharp.Jupyter;
+using Microsoft.Quantum.Simulation.Simulators;
+using Newtonsoft.Json;
+
+namespace Microsoft.Quantum.IQSharp.Kernel
+{
+    /// <summary>
+    ///      Contains the JSON representation of the <see cref="ExecutionPath"/>
+    ///      and the ID of the HTML div that will contain the visualization.
+    /// </summary>
+    public class ExecutionPathVisualizerContent : MessageContent
+    {
+        /// <summary>
+        ///     Initializes <see cref="ExecutionPathVisualizerContent"/> with the
+        ///     given JSON string and HTML div ID.
+        /// </summary>
+        public ExecutionPathVisualizerContent(string json, string id)
+        {
+            this.Json = json;
+            this.Id = id;
+        }
+
+        /// <summary>
+        ///     JSON representation of the <see cref="ExecutionPath"/>.
+        /// </summary>
+        [JsonProperty("json")]
+        public string Json { get; }
```

Comment:
It looks like this will cause objects to get serialized to JSON twice, with the message content containing a JSON string whose value is a JSON object. It may be good to make this something like an `object` to rely on NewtonsoftJson correctly finding the right serializer, or something like `JToken` if you want to serialize it manually before setting it as the payload for this content.